mage_ai/data_preparation/git/__init__.py

```diff
@@ -118,59 +123,67 @@ def wrapper(self, *args, **kwargs):
             with self.repo.git.custom_environment(GIT_SSH_COMMAND=git_ssh_cmd):
                 try:
                     asyncio.run(self.check_connection())
-                except TimeoutError as e:
+                except TimeoutError as te:
                     url = f'ssh://{self.git_config.remote_repo_link}'
                     hostname = urlparse(url).hostname
                     if hostname:
                         cmd = f'ssh-keyscan -t rsa {hostname} >> ~/.ssh/known_hosts'  # noqa: E501
-                        proc = subprocess.Popen(args=cmd, shell=True)
-                        proc.wait()
+                        self._run_command(cmd)
                         asyncio.run(self.check_connection())
                     else:
-                        raise e
+                        raise te
                 func(self, *args, **kwargs)
 
         return wrapper
 
     @_remote_command
-    def reset(self, branch: str = None):
+    def reset(self, branch: str = None) -> None:
         self.origin.fetch()
         if branch is None:
             branch = self.current_branch
         self.repo.git.reset('--hard', f'{self.origin.name}/{branch}')
```

Comment:
it might be helpful to add it here as well.

Reply:
> it might be helpful to add it here as well.

Good catch: added the suggested change.